templates/environment/partials/nat-gateways.yml

```diff
@@ -0,0 +1,30 @@
+{{- range $ind, $cidr := .PrivateSubnetCIDRs}}
+NatGateway{{inc $ind}}Attachment:
+  Type: AWS::EC2::EIP
+  DependsOn: InternetGatewayAttachment
+  Properties:
+    Domain: vpc
```

Comment:
Sorry I know this is corresponding to https://github.com/aws/copilot-cli/pull/1995/files#diff-0d0572768ab2cd6474670c6dffa904880e972f3e3593980a05e6e06386ec07b6R80-R129. But still I'm not fully convinced by this architecture. Have two questions:

1. Why do we use EIP?
2. I don't see any association between EIP and the other resources we create. Is EIP really in used? There's also `EIPAssociation` which seems to be used to associate EIP with a network interface: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-eip-association.html

Reply:
1. Each NAT Gateway needs to have an EIP because it sends traffic to the internet gateway using the EIP address as the source IP address.
2. You use `EIPAssociation` when you have an existing EIP that you want to use, so you associate it with an instance or network interface. Here, for each private subnet, we are creating a new EIP and associating it with a new NAT gateway (by specifying the EIP's `AllocationId`).

Reply:
Oh i guess i missed the `AllocationId` so it's the other way around. Thank you for the explanation!